src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -788,6 +807,50 @@ private void discoverSifting()
         Log.getLogger().info("Finished initiating sifter config");
     }
 
+    /**
+     * Calculate the crusher modes from the config file.
+     */
+    private void discoverEnchantments()
+    {
+        for (final String string : Configurations.gameplay.enchantments)
+        {
+            final String[] split = string.split(":");
+            if (split.length != 4)
+            {
+                Log.getLogger().warn("Invalid enchantment mode setting: " + string);
+                continue;
+            }
+
+            try
+            {
+                final String enchantment = split[1];
+                if (Enchantment.getEnchantmentByLocation(enchantment) == null)
+                {
+                    Log.getLogger().warn("Enchantment: " + enchantment + " doesn't exist!");
+                    continue;
+                }
+
+                final int buildingLevel = Integer.parseInt(split[0]);
+                final int enchantmentLevel = Integer.parseInt(split[2]);
+                final int numberOfTickets = Integer.parseInt(split[3]);
+
+                for (int level = buildingLevel; level <= 5; level++)
+                {
+                    final List<Tuple<String, Integer>> list = enchantments.getOrDefault(level, new ArrayList<>());
+                    for (int i = 0; i < numberOfTickets; i++)
+                    {
+                        list.add(new Tuple<>(enchantment, enchantmentLevel));
+                    }
+                    enchantments.put(level, list);
+                }
+            }
+            catch (final NumberFormatException ex)
+            {
+                Log.getLogger().warn("Invalid integer at pos 1, 3 or 4");
```

Comment:
Check if the enchantment exists here!

Reply:
I do: if (Enchantment.getEnchantmentByLocation(enchantment) == null)

Reply:
Ah okey.